### PR TITLE
Re-commit of what should have been committed

### PR DESCRIPTION
**Pull request by sfcsumb**
Somehow main and dev branches got confused, and code I was playing with got mixed in, and original commit was lost. This is a redo of the original (I hope.)

### CODE REVIEW

File: cst438_gradebook/src/main/java/com/cst438/controllers/AssignmentController.java [hunk 1]
```diff
@@ -29,55 +31,71 @@
...
-		
+		if(assignment.getNeedsGrading() == 0) {
+			assignmentRepository.delete(assignment);			
+		}
+		else {
+			throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Cannot delete, assignment has grades.");
+		}
 	}
 	
 	private Assignment checkAssignment(Assignment assignment, String email) {
```
Comment (denizozgur) [inline cst438_gradebook/src/main/java/com/cst438/controllers/AssignmentController.java:101]: I really like this func. I should implement this to my code too..

File: cst438_gradebook/src/main/java/com/cst438/controllers/EnrollmentController.java [hunk 2]
```diff
@@ -30,11 +30,12 @@ public class EnrollmentController {
...
+		enrollment.setStudentName(enrollmentDTO.studentName);
+		enrollment.setStudentEmail(enrollmentDTO.studentEmail);		
+		enrollment.setCourse(courseRepository.findById(enrollmentDTO.course_id).orElse(null));
+		enrollmentRepository.save(enrollment);
 		
-		//TODO  complete this method in homework 4
-		
-		return null;
-		
+		return enrollmentDTO;
```
Comment (denizozgur) [inline cst438_gradebook/src/main/java/com/cst438/controllers/EnrollmentController.java:39]: I think conventional object to return from a POST request is the response of enrollment repository save function which should be an enrollment object

**Review by denizozgur — DISMISSED** [review]
Code looks clean and functional. Great work !!

**Review by denizozgur — APPROVED** [review]
Code looks clean and functional. Great work Steven

File: cst438_gradebook/src/main/java/com/cst438/controllers/EnrollmentController.java (hunk 2 above)
Comment (sfcsumb) [reply cst438_gradebook/src/main/java/com/cst438/controllers/EnrollmentController.java:39]: The return of enrollmentDTO is what was defined for us in the method definition. Let me know if you have a problem, so far it seemed to work ok for me.

**Pull request merged**